### DISCUSSION TITLE
Unable to build SDK with JDK 8 or 11?

### DISCUSSION THREAD

**Discussion by generate-me12** [open-telemetry/opentelemetry-java#6644] [Q&A] [answered]
I'm new to OpenTelemetry and Gradle. I'm trying to build the sdk with a container with openjdk-8 install but I'm unable to build it. From the versioning, it seems like all artifacts support Java language level 8 or higher. However when I try to build just the sdk with openjdk-8 using `./gradlew sdk:build` 
I get the following error message:
```

FAILURE: Build failed with an exception.
* What went wrong:
A problem occurred configuring project ':buildSrc'.

> Could not resolve all artifacts for configuration ':buildSrc:classpath'.
   > Could not resolve com.diffplug.spotless:spotless-plugin-gradle:6.25.0.
     Required by:
         project :buildSrc > com.diffplug.spotless:com.diffplug.spotless.gradle.plugin:6.25.0
      > Dependency requires at least JVM runtime version 11. This build uses a Java 8 JVM.
* Try:
> Run this build using a Java 11 or newer JVM.
```
When I try to build it with openjdk11, I still get the same error message:
```
FAILURE: Build completed with 2 failures.
1: Task failed with an exception.
-----------
* Where:
Build file '/etc/opentelemetry-java-main/buildSrc/build.gradle.kts' line: 9 
* What went wrong:
JDK 17 is required to build and gradle was unable to detect it on the system.  Please install it and see https://docs.gradle.org/current/userguide/toolchains.html#sec:auto_detection for details on how gradle detects java toolchains.

```
However, I'm able to successfully build it with jdk17. So does OTel Java SDK only support jdk17 or above?

**Comment by jack-berg** [comment, marked as answer]
Artifacts from `opentelemetry-java` are compile / published with [support for java8+](https://github.com/open-telemetry/opentelemetry-java/blob/main/buildSrc/src/main/kotlin/otel.java-conventions.gradle.kts#L76). However, the building the project uses various gradle features which require later versions of java, and thus requires [java 17](https://github.com/open-telemetry/opentelemetry-java/blob/main/buildSrc/build.gradle.kts#L8-L14). 

So java 8+ is required to consume project artifacts, java 17 is required to build from source locally.
